jdaviz/configs/cubeviz/plugins/tests/test_parsers.py

```diff
@@ -209,8 +209,26 @@ def test_numpy_cube(cubeviz_helper):
     assert flux.units == 'ct / pix2'
 
 
+def test_loading_with_mask(cubeviz_helper):
+    # This also tests that spaxel is converted to pix**2
+    custom_spec = Spectrum1D(flux=[[[20, 1], [9, 1]], [[3, 1], [6, np.nan]]] * u.Unit("erg / Angstrom s cm**2 spaxel"),  # noqa
+                             spectral_axis=[1, 2]*u.AA,
+                             mask=[[[1, 0], [0, 0]], [[0, 0], [0, 0]]])
+    cubeviz_helper.load_data(custom_spec)
+
+    uc = cubeviz_helper.plugins['Unit Conversion']
+    uc.spectral_y_type = "Surface Brightness"
+
+    se = cubeviz_helper.plugins['Spectral Extraction']
+    se.function = "Mean"
+    se.extract()
+    extracted = cubeviz_helper.get_data("Spectrum (mean)")
+    assert_allclose(extracted.flux.value, [6, 1])
```

Comment:
Note for my future self so I don't have to do math again:

1. (9 + 3 + 6) / 3 = 6
2. (1 + 1 + 1) / 3 = 1